### ISSUE TITLE
Why do I have a (apparently unused) `/etc/timeshift/timeshift.json` as well as a (apparently used) `/etc/timeshift.json`?

### ISSUE THREAD

**Issue by OwenAR** [teejee2008/timeshift#805]
Timeshift v20.03
Ubuntu 20.04.3

So I seem to have two copies of this config file:
- apparently unused: `/etc/timeshift/timeshift.json`
- apparently used: `/etc/timeshift.json`

and that makes me a bit nervous that I could end up configuring everything correctly,
but then still messing things up by somehow causing the wrong config file to be used for something.

---
It also makes me wonder: `timeshift.json` (the one at whichever location actually ends up getting used) *is* the only config I need to worry about, right?

**Comment by icarter09** [comment]
Timeshift v21.09.1
Linux Mint 20.1 64bit

I only have `/etc/timeshift/timeshift.json, which is being used and works correctly. You can verify which is the one being used looking at the timeshift logs (`/var/log/timeshift`).

A test to try would be to rename or move the `/etc/timeshift.json` file, create a snapshot, then check the given log file to verify the `/etc/timeshift/timeshift.json` file was used.

**Comment by OwenAR** [comment]
[
tldr:
note to any hypothetical future equivalents of me:
use the newer version from the ppa like it says in the readme (`sudo add-apt-repository -y ppa:teejee2008/timeshift`)
not an old version from the default repos of your distro.

(
The rest of this comment is basically just a written record of me rubberduckying myself through this, of no interest to anyone,
but whatever, I wrote it already, so I'll just post it.
)
]

Well I mean, when I said "apparently", I meant like, that's the file that I *had* verified gets read and written under all the conditions I could think to test.

And like, if I just delete both
`/etc/timeshift.json` (henceforth "shallower" to make this easier to read)
and
`/etc/timeshift/timeshift.json` (henceforth "deeper")
the only one that gets regenerated (by running `timeshift-gtk`) is "shallower".

And if I delete "shallower"
but leave "deeper"
then it'll read from "deeper"
but write to "shallower".

But clearly, "deeper" must get written to under *some* conditions,
otherwise it wouldn't've existed in the first place,
except I can't figure out what those conditions are,
which makes me worry that there might be mysterious conditions under which it would also *read* from "deeper",
even when "shallower" already exists and I *expected* it to read from that...

... also, under certain conditions I've found the gui silently modifying the config file (ie "shallower") when I just open "settings" and then close it again without touching anything.

---
Okay, I realized that I'm using an older version of timeshift that comes in the default repos for the LTS I'm on
so I just made it install from the ppa instead like it says in the readme (`sudo add-apt-repository -y ppa:teejee2008/timeshift`)
so now I've got v21.09.1,
and I'll poke around and see if that changes things...

Well, I had entirely deleted the dir
`/etc/timeshift/`
(
since it didn't seem to be writing anything to there,
and only ever read from it if "shallower" wasn't already present
)
|
but installing the new version regenned the dir
`/etc/timeshift/`
but rather than containing another version of `timeshift.json`,
it now contains `default.json`
which I guess makes sense of what the purpose of things in the old version was *supposed* to be.

It still *read* my old config from "shallower"
but then it *wrote* to "deeper" instead
(creating it in the process)

... and it automatically deleted "shallower" when it did so, apparently.

So my install is now reading and writing `/etc/timeshift/timeshift.json` ("deeper"),
and "shallower" no longer exists at all.

... yeah, things are finally making sense now.

---
Alright, well, I *think* that resolves this issue for me
(of exactly which config file it's reading/writing).

(I *think* it also fixed an unrelated issue of filter-reordering done through drag-and-drop in the gui not getting written to the file (which I found mentioned in #626).)

It's still silently modifying the config file under certain conditions,
but that's a separate issue.
